### DISCUSSION TITLE
About VAE or GVAE decoder

### DISCUSSION THREAD

**Discussion by EimhinLiu** [pyg-team/pytorch_geometric#9336] [Q&A]
Dear pyg users and developers,
Is there a way other than inner product to reconstruct the node-to-edge representation here?
If so, please let me know the paper or the exact method.

**Comment by rusty1s** [comment]
Inner product is certainly the most common since it allows you to do Maximum Inner Product Search (MIPS) on node embeddings. You can also use other k-NN tools, e.g., based on Euclidean distance). Alternatively, you can have a parametrized decoder, e.g., an MLP on source and destination node embeddings that maps to a link probability, but this is expensive to compute for O(N^2) pairs.
